### PR TITLE
DEBT-16 | Added all card functionality

### PR DESCRIPTION
**Pull request by wdonray**
[DEBT-16](https://donrayxwilliams.atlassian.net/browse/DEBT-16?atlOrigin=eyJpIjoiYTM0ZGNlODc1M2ZkNDNjMTg5NTRhY2YzMTllYTE4ODciLCJwIjoiaiJ9)

### CODE REVIEW

File: components/DebtCards/components/update-debt-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { DebtContext, raiseError, updateDebt } from '@/util';
+
+enum Titles {
+  name = 'Rename',
+  type = 'Change Type',
+}
+
+enum SubmitButton {
+  name = 'Rename',
+  type = 'Change'
```
Comment (chayboats) [inline components/DebtCards/components/update-debt-modal.tsx]:
```suggestion
  type = 'Change',
```

File: util/context/debt-context.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  name = 'name',
+  type = 'type',
+}
+
+export interface DebtContextInterface {
+  selectedDebt: IDebt | null,
+  isUserAuthenticated: boolean,
+  userId: string | undefined,
+  debtList: IDebt[] | undefined,
+  localDebtList: IDebt[] | undefined
```
Comment (chayboats) [inline util/context/debt-context.ts]:
```suggestion
  localDebtList: IDebt[] | undefined,
```

File: util/context/debt-context.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  valueToUpdate: UpdateDebtValue,
+  handleValueToUpdate: (value: UpdateDebtValue) => void,
+  handleAddLocalDebt: (debt: IDebt) => void,
+  handleUpdateLocalDebt: (debt: IDebt) => void,
+  handleDeleteLocalDebt: (id: string) => void,
+  handleSelectedDebt: (debt: IDebt | null) => void
```
Comment (chayboats) [inline util/context/debt-context.ts]:
```suggestion
  handleSelectedDebt: (debt: IDebt | null) => void,
```

File: types/debt.d.ts [hunk 4]
```diff
@@ -32,4 +32,9 @@ export interface IUpdateDebtInput {
   apr?: number
   payment?: number
   userDebtId: string
+}
+
+export interface IDeleteDebtInput {
+  id: string
```
Comment (chayboats) [inline types/debt.d.ts]: Add semicolons everywhere.

**Review by chayboats — APPROVED** [review]
LGTM

**Review by devinbroussard — APPROVED** [review]

**Pull request merged**